### PR TITLE
Added DiscoveryClient

### PR DESCRIPTION
**Pull request by weitzhandler**
#66

### CODE REVIEW

File: source/IdentityModel.Net45/IdentityModel.Net45.csproj [hunk 1]
```diff
@@ -30,14 +30,16 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```
Comment (leastprivilege) [inline source/IdentityModel.Net45/IdentityModel.Net45.csproj]: Was that necessary? Why change to the latest version - this will force everyone to update.

Comment (weitzhandler) [reply source/IdentityModel.Net45/IdentityModel.Net45.csproj]: Reverted. Compiled OK.
See new commit: 9c076482822239d686bbfa3af8580155031fb674

**Comment by leastprivilege** [comment]
Cool. thank you!

**Comment by leastprivilege** [comment]
I merged it - but there is an issue -

the build is not setup for C# 6/7 - try running build.ps1 yourself to see.

I know there is a way to change that - but I can't find it right now - if you want to give it a try - this would speed up the nuget publishing.

**Comment by weitzhandler** [comment]
It built on my machine, is there anything I should have been aware of? I might have made use of C#7 code just of habit.
In fact I don't remember which code is C# 6/7. Can I force my compiler to that, to make sure it compiles everywhere?

**Comment by leastprivilege** [comment]
does the build script run on your machine?

**Comment by leastprivilege** [comment]
it will always build in VS - but the script is the important one for publishing

**Comment by weitzhandler** [comment]
Can you direct me how to test with the build script?

**Comment by leastprivilege** [comment]
run build.ps1 in the root dir via powershell

**Comment by weitzhandler** [comment]
ok will do that now and let you know

**Comment by weitzhandler** [comment]
I can't find the file `build.ps1` in my local repo.

**Comment by leastprivilege** [comment]
The reason the build script fails is because it picks up the wrong version of msbuild - but I can't remember the "right" way to fix that.

**Comment by leastprivilege** [comment]
https://github.com/IdentityModel/IdentityModel/blob/dev/build.ps1

**Comment by weitzhandler** [comment]
I get a gazillion errors:
https://pastebin.com/raw/WUfJLxaD

**Comment by weitzhandler** [comment]
I'm on VS 2017, could it be the newer project model? I didn't get any upgrade log etc. when opening solution.
But I can spot some C# errors too.

**Comment by weitzhandler** [comment]
What C# version should be supported? Let's enforce this in project settings, then I'll walk thru all the errors and downgrade the code.

**Comment by leastprivilege** [comment]
Well - originally IdentityModel is from the C# 5 era. We didn't want to force anyone to use a newer version of VS to work with it. The situation is slightly different today.

It's a matter of changing the build script to use the version of MsBuild that understands the newer C# - I just don't have the time right now.

So if you want to speed up the process - research that.

**Comment by weitzhandler** [comment]
Maybe will dig into it a bit later.

**Comment by weitzhandler** [comment]
Hopefully will dig into it even before you do...

**Pull request merged**